### ISSUE TITLE
ManyToMany relation doesn't put an empty array when the relation is empty

### ISSUE THREAD

**Issue by blue112** [typeorm/typeorm#175]
When you make a ManyToMany relation with a join table and the relation doesn't return any result, the resulting field is set to undefined.

It would help if it would be set to an empty array, since it would prevent handling a special case.

**Comment by paistipoikka** [comment]
This also happens on OneToMany. For example.

```
return this.ormRepository.createQueryBuilder("organisation")
            .leftJoinAndSelect("organisation.permissions", "permissions")
            .andWhere("organisation.id = :id")
            .setParameter("id", id)
            .getOne();
```

If no permissions exist then the permissions property is undefined.

**Comment by pleerock** [comment]
fixed in `0.0.7-alpha.11`

**Comment by leonelgasparrini** [comment]
This still doesn't work on version
```
"typeorm": "^0.2.25",
```
